solr/contrib/ltr/src/test/org/apache/solr/ltr/feature/impl/TestLambdaMARTModel.java

```diff
@@ -127,60 +127,121 @@ public void lambdaMartTestExplain() throws Exception {
-        "lambdamart_features.json");
+    final ModelException expectedException = 
+        new ModelException("Model type does not exist org.apache.solr.ltr.ranking.LambdaMARTModel");
+    try {
+        createModelFromFiles("lambdamart_model_no_weight.json",
+            "lambdamart_features.json");
+        fail("unexpectedly got here instead of catching "+expectedException);
+    } catch (ModelException actualException) {
+      assertEquals(expectedException.toString(), actualException.toString());
+    }
   }
 
-  @Test(expected = ModelException.class)
+  @Test
   public void lambdaMartTestNoTree() throws Exception {
-    createModelFromFiles("lambdamart_model_no_tree.json",
-        "lambdamart_features.json");
+    final ModelException expectedException = 
+        new ModelException("Model type does not exist org.apache.solr.ltr.ranking.LambdaMARTModel");
+    try {
+        createModelFromFiles("lambdamart_model_no_tree.json",
+            "lambdamart_features.json");
+        fail("unexpectedly got here instead of catching "+expectedException);
+    } catch (ModelException actualException) {
+      assertEquals(expectedException.toString(), actualException.toString());
+    }
   }
 
-  @Test(expected = SolrException.class)
+  @Test
   public void lambdaMartTestNoFeatures() throws Exception {
-    createModelFromFiles("lambdamart_model_no_features.json",
-        "lambdamart_features.json");
+    final SolrException expectedException = 
+        new SolrException(SolrException.ErrorCode.BAD_REQUEST,
+            "Missing mandatory field features");
+    try {
+        createModelFromFiles("lambdamart_model_no_features.json",
+            "lambdamart_features.json");
+        fail("unexpectedly got here instead of catching "+expectedException);
+    } catch (SolrException actualException) {
+      assertEquals(expectedException.toString(), actualException.toString());
+    }
   }
 
-  @Test(expected = ModelException.class)
+  @Test
   public void lambdaMartTestNoRight() throws Exception {
-    createModelFromFiles("lambdamart_model_no_right.json",
-        "lambdamart_features.json");
-
+    final ModelException expectedException = 
+        new ModelException("Model type does not exist org.apache.solr.ltr.ranking.LambdaMARTModel");
+    try {
+        createModelFromFiles("lambdamart_model_no_right.json",
+            "lambdamart_features.json");
+        fail("unexpectedly got here instead of catching "+expectedException);
+    } catch (ModelException actualException) {
+      assertEquals(expectedException.toString(), actualException.toString());
+    }
   }
 
-  @Test(expected = ModelException.class)
+  @Test
   public void lambdaMartTestNoLeft() throws Exception {
-    createModelFromFiles("lambdamart_model_no_left.json",
-        "lambdamart_features.json");
+    final ModelException expectedException = 
+        new ModelException("Model type does not exist org.apache.solr.ltr.ranking.LambdaMARTModel");
```

Comment:
This is a poor error message. Should say "Model .... could not be built, Missing left"